service/matching/task_queue_manager.go

```diff
@@ -417,6 +419,12 @@ func (c *taskQueueManagerImpl) AddTask(
 		c.liveness.markAlive()
 	}
 
+	if c.taskQueueID.VersionSet() == dlqVersionSet {
+		// If this is the versioned task dlq, skip straight to spooling the task since we can't
+		// have any pollers.
+		return false, c.SpoolTask(params)
```

Comment:
Is this just an optimization to skip trying to sync match because there will never be any pollers on the DLQ? If so, is it worth the added complexity?

Reply:
Sorta.. I didn't really think of it as an optimization, more as a simplification: I don't have to consider whether the rest of this function does a sensible thing with the dlq since it won't even get there.

But reading again, I think actually we do need the errRemoteSyncMatchFailed check, since dlq tasks could still get forwarded. I'll change it.

I think we do want to skip sync match, though, not for the sync match part but for the forwarding